lib/install/management/configure.go

```diff
@@ -98,49 +102,150 @@ func (d *Dispatcher) Configure(vch *vm.VirtualMachine, conf *config.VirtualConta
+		d.op.Debugf("No inventory update completed for %s", name)
+		return nil
+	}
 
-		d.deleteUpgradeImages(ds, settings)
-		d.retryDeleteSnapshotByRef(newSnapshotRef, snapshotName, conf.Name)
+	// concatenate w/vch name so we can identify if failure
+	tempName := fmt.Sprintf("%s-%s", name, uid.New().Truncate().String())
+	// create target folder with random name for VCH Inventory
+	target, err := d.session.VMFolder.CreateFolder(d.op, tempName)
+	if err != nil {
+		d.op.Errorf("Unable to create target folder during inventory update: %s", err)
+		return err
+	}
 
-		// return the error message for upgrade
+	rp := compute.NewResourcePool(d.op, d.session, d.vchPool.Reference())
+	children, err := rp.GetChildrenVMs(d.op)
+	if err != nil {
+		d.op.Errorf("Unable to locate children during inventory update: %s", err)
 		return err
 	}
 
-	// compatible with old version's upgrade snapshot name
-	if oldSnapshot != nil && (vm.IsConfigureSnapshot(oldSnapshot, ConfigurePrefix) || vm.IsConfigureSnapshot(oldSnapshot, UpgradePrefix)) {
-		d.retryDeleteSnapshotByRef(&oldSnapshot.Snapshot, oldSnapshot.Name, conf.Name)
+	var vicVMs []types.ManagedObjectReference
+	for _, c := range children {
+		// This will make a vSphere call for each VM in the RP - this might
+		// prove to be slow and require optimization.  We do this in multiple
+		// places, so ideally provide a central func
+		vic, err := d.isContainerVM(c)
+		if err != nil {
+			// A failure occurred attempting to get the guestInfo which is needed to determine
+			// if the VM is a container.  To avoid missing a containerVM we will fail the update.
+			return err
+		}
+
+		if vic {
+			vicVMs = append(vicVMs, c.Reference())
+		}
+	}
+	// add the appliance for move
+	vicVMs = append(vicVMs, d.appliance.Reference())
+
+	d.op.Debugf("Moving Children(%d) into the folder", len(vicVMs))
+	_, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
+		return target.MoveInto(d.op, vicVMs)
+	})
+
+	// Folder.MoveInto is atomic for each VM move, so it could fail with only part of
+	// the inventory being moved.  This would not cause an issue with vic functionality
+	// or future vic-machine operations.
+	if err != nil {
+		d.op.Warnf("Unable to move all VCH VMs into folder(%s): %s", err)
+		d.op.Warnf("VCH VMs not residing in the folder will still function correctly")
+	} else {
+		d.op.Debugf("Children(%d) moved into folder", len(vicVMs))
+	}
+	_, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
+		return target.Rename(d.op, d.vmPathName)
+	})
+	if err != nil {
+		// Much like folder.MoveInto this will not impact the ability for vic to operate, so
+		// only a warning is issued.
+		d.op.Warnf("Unable to rename folder(%s) to %s during inventory update: %s", tempName, name, err)
+		d.op.Warnf("Folder naming will not interfere with VCH operation, rename via VC if desired")
```

Comment:
We don't make any assumptions about folder name?

Reply:
No - we don't store the folder name as part of config (unlike RP) so on portlayer init we grab the current folder...whatever it might be...